lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -548,6 +556,17 @@ defmodule ExUnit.AssertionsTest do
       "\n  \"bar\"" = error.message
   end
 
+  test "assert raise with an exception with bad message/1 implementation" do
+    assert_raise BrokenError, fn ->
+      raise BrokenError
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Got exception ExUnit.AssertionsTest.BrokenError but it failed to produce a message with:" <>
+      "\n\n** (RuntimeError) error" <>
+      "\n    test/ex_unit/assertions_test.exs:13: ExUnit.AssertionsTest.BrokenError.message/1" <> _ = error.message
```

Comment:
Please don't use fixed line numbers as it can break in future when changing other parts of the file :)